### PR TITLE
feat(flatbuffers): add Dodd-Frank mid price properties

### PR DESCRIPTION
**Pull request by ljm-reactivemarkets**
We need to support mid prices on Quotes, MarketDataSnapshots and Execution reports

SD-4383

### CODE REVIEW

File: flatbuffers/MarketDataSnapshot.fbs [hunk 1]
```diff
@@ -56,4 +56,8 @@ table MarketDataSnapshot {
     bid: [MDEntry];
     /// Offer market data entries.
     offer: [MDEntry];
+    /// Dodd-Frank mid price.
+    mid_price: double;
+    /// Swaps only, far leg Dodd-Frank mid price.
+    far_mid_price: double;
```
Comment (markmcdowell) [inline flatbuffers/MarketDataSnapshot.fbs:62]: Don't think we need this at the moment, no swaps over ESP.

Comment (ljm-reactivemarkets) [reply flatbuffers/MarketDataSnapshot.fbs]: Fair, was thinking it is OK to add as we will use it eventually, but I see all other far leg related fields are missing too. Removed

File: flatbuffers/Quote.fbs [hunk 2]
```diff
@@ -74,4 +74,8 @@ table Quote {
     bid: [QuoteEntry];
     /// Offer quote entries.
     offer: [QuoteEntry];
+    /// Dodd-Frank mid price.
+    mid_price: double;
+    /// Swaps only, far leg Dodd-Frank mid price
+    far_mid_price: double;
```
Comment (markmcdowell) [inline flatbuffers/Quote.fbs:80]: Do we need to confirm if we'll have a far mid? It it's in swap points it'll be a single mid.

Comment (markmcdowell) [reply flatbuffers/Quote.fbs:80]: Checked SCB and they provide mid per leg + mid in swap points. So it would be a single mid_price.

Comment (ljm-reactivemarkets) [reply flatbuffers/Quote.fbs:80]: We did raise that in the initial call last week, and the answer was yes, a mid price for each leg.
I also see backend are proceeding with mid price and far mid price
https://github.com/reactivemarkets/schema/pull/1159

**Review by markmcdowell — APPROVED** [review]

**Pull request merged**